var/spack/repos/builtin/packages/py-ipdb/package.py

```diff
@@ -26,8 +31,8 @@ class PyIpdb(PythonPackage):
     # However additional dependencies added below were found via testing.
     depends_on("py-setuptools", type="build")
     # ipdb needs iPython and others available at runtime
-    depends_on("py-ipython@0.10.2:", type=("build", "link"))
-    depends_on("py-traitlets", type=("build", "link"))
-    depends_on("py-six", type=("build", "link"))
-    depends_on("py-pexpect", type=("build", "link"))
-    depends_on("py-prompt-toolkit", type=("build", "link"))
+    depends_on("py-ipython@0.10.2:", type=("build", "link", "run"))
```

Comment:
For the latest version of ipdb, this should look like:
```python
depends_on("py-ipython@7.16.3:7.16", when="@0.13.1: ^python@7.6:3.10", type=("build", "run"))
depends_on("py-ipython@7.31.1:", when="@0.13.1: ^python@3.11:", type=("build", "run"))
```
You'll need to see what the requirements were for older versions and adjust/add new lines accordingly